torch/csrc/jit/python/init.cpp

```diff
@@ -694,6 +694,29 @@ void initJITBindings(PyObject* module) {
             std::tie(data, size) = self.getRecord(key);
             return py::bytes(reinterpret_cast<const char*>(data.get()), size);
           })
+      .def(
+          "get_storage_from_record",
+          [](PyTorchStreamReader& self,
+             const std::string& key,
+             size_t numel,
+             py::object dd) {
+            at::DataPtr data;
+            size_t size;
+            std::tie(data, size) = self.getRecord(key);
+            auto scalar_type =
+                reinterpret_cast<THPDtype*>(dd.ptr())->scalar_type;
```

Comment:
nit: Can this be cast with a safer cast? static_cast perhaps? what type is `dd.ptr()` before cast?